torchgeo/datamodules/ssl4eo_benchmark_landsat.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SSL4EO datamodule."""
+
+from typing import Any
+
+import kornia.augmentation as K
+
+from ..datasets import SSL4EOLBenchmark
+from ..transforms import AugmentationSequential
+from .geo import NonGeoDataModule
+
+
+class SSL4EOLBenchmarkDataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the SSL4EO-L dataset.
+
+    .. versionadded:: 0.5
+    """
+
+    mean = 0
+    std = 255
+
+    def __init__(
+        self, batch_size: int = 64, num_workers: int = 0, **kwargs: Any
+    ) -> None:
+        """Initialize a new SSL4EOLDownstreamDataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            num_workers: Number of workers for parallel data loading.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.SSL4EOLBenchmark`.
+        """
+        super().__init__(SSL4EOLBenchmark, batch_size, num_workers, **kwargs)
+
+        self.train_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            K.CenterCrop(224),
+            data_keys=["image", "mask"],
+        )
+        self.val_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            K.CenterCrop(224),
+            data_keys=["image", "mask"],
+        )
+        self.test_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            K.CenterCrop(224),
+            data_keys=["image", "mask"],
+        )
+        self.predict_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            K.CenterCrop(224),
+            data_keys=["image", "mask"],
+        )
+
+    def setup(self, stage: str) -> None:
+        """Set up datasets.
+
+        Args:
+            stage: Either 'fit', 'validate', 'test', or 'predict'.
+        """
+        self.dataset = SSL4EOLBenchmark(**self.kwargs)
```

Comment:
I think we want the default base class function here. We need to pass in the split based on stage.

Reply:
It could be because the fake data doesn't have enough examples to do the current split ratio without one split ending up with 0 images? Remind me to improve the error messages to clarify why the data loaders don't work, right now the message is really generic.

Reply:
I created 3 data examples and monkeypatch the split as [1/3,1/3, 1/3], it was always working before.

Reply:
You only monkey-patched the dataset, not the datamodule. Maybe we could make those params of the datamodule/dataset? Also, with a batch size of 2, you'll need at least 6 data examples with 1/3 splits.

Reply:
batch_size is always a "max" or no? So it would at most take that many samples but if there is only less available it will  just take those.

Reply:
should I just create more fake samples then so the test_segmentation script always has enough samples?

Reply:
Yes, but 15% of 2 is 0.